### PR TITLE
fix Hec 6.1, cleaner approach to read hec file

### PR DESCRIPTION
**Pull request by vcloarec**

### CODE REVIEW

File: mdal/frmts/mdal_hec2d.cpp [hunk 1]
```diff
@@ -761,18 +760,30 @@ std::unique_ptr<MDAL::Mesh> MDAL::DriverHec2D::load( const std::string &resultsF
...
 
     std::vector<std::string> flowAreaNames;
-    if ( oldFormat )
-      flowAreaNames = read2DFlowAreasNamesOld( gGeom2DFlowAreas );
-    else
-      flowAreaNames = read2DFlowAreasNames505( gGeom2DFlowAreas );
+
+    try
+    {
+      flowAreaNames = read2DFlowAreasNamesFromNameDataset( gGeom2DFlowAreas );
```
Comment (PeterPetrik) [inline mdal/frmts/mdal_hec2d.cpp:770]: can we change this read2DFlowAreasNamesFromNameDataset() and read2DFlowAreasNamesFromAttributesDataset() to just return empty flowAreaNames on error and not throw? 

it is generally not advisable to use try-catch for expected code-flow and not error situation.

**Review by PeterPetrik — COMMENTED** [review]
this looks ok

File: mdal/frmts/mdal_hec2d.cpp (hunk 1 above)
Comment (vcloarec) [reply mdal/frmts/mdal_hec2d.cpp]: done

**Review by PeterPetrik — APPROVED** [review]

**Comment by vcloarec** [comment]
windows test is now fix and lives in 2022.

**Pull request merged**